ivy/functional/backends/paddle/experimental/layers.py

```diff
@@ -245,7 +246,39 @@ def fft(
     n: Union[int, Tuple[int]] = None,
     out: Optional[paddle.Tensor] = None,
 ) -> paddle.Tensor:
-    raise IvyNotImplementedException()
+    if not isinstance(dim, int):
+        raise ivy.utils.exceptions.IvyValueError(
+            f"Expecting <class 'int'> instead of {type(dim)}"
+        )
+
+    if n is None:
+        n = x.shape[dim]
+
+    if n < -len(x.shape) or n >= len(x.shape):
+        raise ivy.utils.exceptions.IvyValueError(
+            f"Invalid dim {dim}, expecting a value ranging from {-len(x.shape)} to {len(x.shape)-1}"
+        )
+
+    if not isinstance(n, int):
+        raise TypeError(
+            f"Expecting int type for 'n', instead of {type(n)}"
+        )
+
+    if n <= 1:
+        raise ivy.utils.exceptions.IvyValueError(
+            f"Invalid number of data points {n}, expecting more than 1"
+        )
+        
+    valid_norm_modes = ["backward", "ortho", "forward"]
+    if norm not in valid_norm_modes:
+        raise ivy.utils.exceptions.IvyValueError(f"Unrecognized normalization mode {norm}, expecting one of {valid_norm_modes}")
+
+    if x.dtype in [paddle.int64, paddle.float64, paddle.complex128]:
+        out_dtype = paddle.complex128
+    else:
+        out_dtype = paddle.complex64
+
+    return paddle.fft.fft(x, n, dim, norm=norm).astype(out_dtype)
```

Comment:
```suggestion
if not isinstance(dim, int):
        raise IvyValueError(
            f"Expecting <class 'int'> instead of {type(dim)}"
        )

    if n is None:
        n = x.shape[dim]

    if dim < -x.ndim or dim >= x.ndim:
        raise IvyValueError(
            f"Invalid dim {dim}, expecting a value ranging from {-x.ndim} to {x.ndim-1}"
        )

    if not isinstance(n, int):
        raise TypeError(
            f"Expecting int type for 'n', instead of {type(n)}"
        )

    if n <= 1:
        raise IvyValueError(
            f"Invalid number of data points {n}, expecting more than 1"
        )

    valid_norm_modes = ["backward", "ortho", "forward"]
    if norm not in valid_norm_modes:
        raise IvyValueError(f"Unrecognized normalization mode {norm}, expecting one of {valid_norm_modes}")

    if x.dtype in [paddle.int64, paddle.float64, paddle.complex128]:
        x = x.cast(paddle.complex128)
    else:
        x = x.cast(paddle.complex64)

    return paddle.fft.fft(x, n, dim, norm=norm)
```